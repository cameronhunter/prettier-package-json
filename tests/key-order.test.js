const { format } = require('../src');

test('It orders keys using the default key order', () => {
  const json = {
    description: 'Description',
    version: '0.0.0',
    name: 'Test'
  };

  expect(format(json)).toMatchSnapshot();
});

test('It orders keys using the specified key order', () => {
  const json = {
    description: 'Description',
    version: '0.0.0',
    name: 'Test'
  };

  expect(format(json, { keyOrder: ['name', 'description', 'version'] })).toMatchSnapshot();
});

test('It orders unspecified keys alphabetically at the end', () => {
  const json = {
    unknownPropertyB: false,
    unknownPropertyA: true,
    description: 'Description',
    version: '0.0.0',
    name: 'Test'
  };

  expect(format(json, { keyOrder: ['name', 'version', 'description'] })).toMatchSnapshot();
});

test('It orders keys using a custom key order function', () => {
  const json = {
    description: 'Description',
    version: '0.0.0',
    name: 'Test'
  };

  const keyOrder = (keyA, keyB) => {
    if (keyA === keyB) {
      return 0;
    } else {
      return keyA < keyB ? -1 : 1;
    }
  };

  expect(format(json, { keyOrder })).toMatchSnapshot();
});

test('It orders scripts in alphabetical order, keeping pre and post scripts beside their counterparts', () => {
  const json = {
    scripts: {
      test: 'test',
      pretest: 'pretest',
      version: 'version',
      postversion: 'postversion',
      build: 'build'
    }
  };

  expect(format(json)).toMatchSnapshot();
});
