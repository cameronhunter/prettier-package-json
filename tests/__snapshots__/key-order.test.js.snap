// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It orders keys using the default key order 1`] = `
"{
  \\"name\\": \\"Test\\",
  \\"description\\": \\"Description\\",
  \\"version\\": \\"0.0.0\\"
}
"
`;

exports[`It orders keys using the specified key order 1`] = `
"{
  \\"name\\": \\"Test\\",
  \\"version\\": \\"0.0.0\\",
  \\"description\\": \\"Description\\"
}
"
`;

exports[`It orders scripts in alphabetical order, keeping pre and post scripts beside their counterparts 1`] = `
"{
  \\"scripts\\": {
    \\"build\\": \\"build\\",
    \\"pretest\\": \\"pretest\\",
    \\"test\\": \\"test\\",
    \\"version\\": \\"version\\",
    \\"postversion\\": \\"postversion\\"
  }
}
"
`;
